packages/local-environment/src/network.ts

```diff
@@ -32,31 +32,44 @@ export abstract class Network {
   bridgeContracts?: BridgeContracts;
   bridgeGui?: AppConfig;
 
+  docker: Dockerode;
+
   blockTime: number;
 
   updater: string;
   watcher: string;
   recoveryManager: string;
   weth: string;
 
+  rpc: string[];
+
+  abstract up(): Promise<void>;
+  abstract down(): Promise<void>;
+  abstract isConnected(): Promise<boolean>;
   abstract get specs(): NetworkSpecs;
   abstract get rpcs(): string[];
   abstract get config(): ContractConfig;
   abstract get bridgeConfig(): BridgeConfiguration;
 
-  abstract up(): Promise<void>;
-  abstract down(): Promise<void>;
-  abstract isConnected(): Promise<boolean>;
   abstract getJsonRpcSigner(
     addressOrIndex: string | number
   ): ethers.providers.JsonRpcSigner;
   abstract getJsonRpcProvider(): ethers.providers.JsonRpcProvider;
+  abstract setWETH(wethAddress: string): string;
+  abstract deployWETH(): Promise<string>;
+  abstract addKeys(...keys: Key[]): void;
+  abstract deployToken(
+    contractFactory: ethers.ContractFactory,
+    from: string | ethers.providers.JsonRpcSigner,
+    ...args: string[]
+  ): Promise<ethers.Contract>;
 
   constructor(
     name: string,
     domainNumber: number,
     chainId: number,
-    blockTime = 10000
+    docker: Dockerode,
+    forkUrl?: string
```

Comment:
Same as domain comment. `Network` is lower level stuff. Not the place for docker